pkg/ddc/thin/referencedataset/runtime.go

```diff
@@ -91,20 +96,39 @@ func (e *ReferenceDatasetEngine) getRuntimeInfo() (base.RuntimeInfoInterface, er
 	e.Log.Info("Setup with dataset done", "exclusive", e.runtimeInfo.IsExclusive())
 
 	return e.runtimeInfo, nil
-
 }
 
+// getMountedRuntimeInfo get mountedRuntimeInfo from dataset.
+// If could not get dataset, getMountedRuntimeInfo try to get mountedRuntimeInfo from runtime status.
+// And if dataset is deleted and no status.mounts, it returns nil, nil to continue runtimeDeletion.
 func (e *ReferenceDatasetEngine) getMountedRuntimeInfo() (base.RuntimeInfoInterface, error) {
 	if e.mountedRuntimeInfo != nil {
 		return e.mountedRuntimeInfo, nil
 	}
 
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
+	// if err is not found, try to get mountedRuntimeInfo from runtime status, don't return here.
+	if err != nil && utils.IgnoreNotFound(err) != nil {
+		return e.mountedRuntimeInfo, err
+	}
+
+	runtime, err := e.getRuntime()
 	if err != nil {
 		return e.mountedRuntimeInfo, err
 	}
 
-	mountedNameSpacedNames := base.GetMountedDatasetNamespacedName(dataset)
+	var mountedNameSpacedNames []types.NamespacedName
+	if dataset != nil {
+		// get mountedRuntimeInfo from dataset first
+		mountedNameSpacedNames = base.GetPhysicalDatasetFromMounts(dataset.Spec.Mounts)
+	} else if runtime.Status.Mounts != nil && len(runtime.Status.Mounts) != 0 {
```

Comment:
No need to check `runtime.Status.Mounts != nil` here because `len(nil) == 0`. We can remove `runtime.Status.Mounts != nil` to avoid code redundancy

Reply:
Done.